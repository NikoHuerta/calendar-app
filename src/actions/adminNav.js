import { types } from "../types/types";

export const setNavigation = ( page ) => ({
    type: types.adminNavigation,
    payload: page
});