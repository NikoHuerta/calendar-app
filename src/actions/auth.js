import Swal from "sweetalert2";

import { fetchSinToken, fetchAxios } from "../helpers/fetch"
import { types } from "../types/types";


export const startLogin = (email, password) => {
    return async ( dispatch ) => {

        // const resp = await fetchSinToken('auth', { email, password }, 'POST');
        // const body = await resp.json();
        const resp = await fetchAxios('auth', { email, password }, 'POST');
        const { data: body } = resp;

        if(body.ok){
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime() );

            dispatch( login({ 
                    uid: body.uid,
                    name: body.name
                }) );
        } else {

            const errorMsg = body.errors ? Object.values( body.errors )[0].msg : body.msg;
            Swal.fire('Error', errorMsg, 'error');
        }
    }
}

export const startRegister = (email, password, name) => {
    return async (dispatch) => {

        // const resp = await fetchSinToken('usuarios', { name, email, password, rol: 'USER_ROLE' }, 'POST');
        // const body = await resp.json();
        const resp = await fetchAxios('usuarios', { name, email, password, rol: 'USER_ROLE' }, 'POST');
        const { data: body } = resp;

        if(body.ok){
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime() );

            dispatch( login({ 
                uid: body.uid,
                name: body.name
            }) );
            
        } else {
            
            const errorMsg = body.errors ? Object.values( body.errors )[0].msg : body.msg;
            Swal.fire('Error', errorMsg, 'error');
        }
    }
}

export const startGoogleLogin = (idToken) => {
    return async (dispatch) => {

        // const resp = await fetchSinToken('auth/google', { id_token: idToken }, 'POST');
        // const body = await resp.json();
        const resp = await fetchAxios('auth/google', { id_token: idToken }, 'POST');
        const { data: body } = resp;

        if(body.ok){
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime() );

            dispatch( login({ 
                uid: body.usuario.uid,
                name: body.usuario.name
            }) );

        } else {
            
            const errorMsg = body.errors ? Object.values( body.errors )[0].msg : body.msg;
            Swal.fire('Error', errorMsg, 'error');
        }
    }
}

export const startFacebookLogin = (idToken, name, email) => {
    return async (dispatch) => {
        // const resp = await fetchSinToken('auth/facebook', { id_token: idToken, name, email }, 'POST');
        // const body = await resp.json();

        const resp = await fetchAxios('auth/facebook', { id_token: idToken, name, email }, 'POST');
        const { data: body } = resp;


        if(body.ok){
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime() );

            dispatch( login({ 
                uid: body.usuario.uid,
                name: body.usuario.name
            }) );

        } else {
            
            const errorMsg = body.errors ? Object.values( body.errors )[0].msg : body.msg;
            Swal.fire('Error', errorMsg, 'error');
        }
    }
}

export const startChecking = () => {
    return async (dispatch) => {
        const resp = await fetchAxios('auth/renew', undefined,'GET',undefined, localStorage.getItem('token'));
        const { data: body } = resp;

        if(body.ok){

            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime() );

            dispatch( login({ 
                uid: body.uid,
                name: body.name
            }) );

        } else {
            dispatch( checkingFinish() );
        }

    }
}

const checkingFinish = () => ({ type: types.authCheckingFinish });

const login = ( user ) => ({
    type: types.authLogin,
    payload: user
});

const logout = () => ({ type: types.authLogout });

export const startLogout = () => {
    return( dispatch ) => {
        localStorage.clear();
        dispatch( logout() );
    }
}
