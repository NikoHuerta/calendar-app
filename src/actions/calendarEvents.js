import { types } from "../types/types"

export const eventAddNew = (event) => ({
    type: types.eventAddNew,
    payload: event
})

export const eventSetActive = (event) => ({
    type: types.eventSetActive,
    payload: event
});

export const eventRemoveActive = () => ({ type: types.eventRemoveActive });

export const eventUpdated = (event) => ({
    type: types.eventUpdated,
    payload: event
}); 

export const eventDeleted = () => ({ type: types.eventDeleted });

export const eventSelectedDate = (dates) => ({ 
    type: types.eventSelectedDate,
    payload: dates
});

export const eventRemoveSelectedDate = () => ({ type: types.eventRemoveSelectedDate });