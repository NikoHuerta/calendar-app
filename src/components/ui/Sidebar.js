import React from 'react'
import { useDispatch } from 'react-redux';
import { setNavigation } from '../../actions/adminNav';

export const Sidebar = () => {

const dispatch = useDispatch();

const handleClick = (e) => {
    e.preventDefault();
    dispatch(setNavigation({page: e.currentTarget.value}));
}

  return (
    <nav className='col-md-2 d-md-block bg-dark text-light sidebar vh-100'>
        <div className='position-sticky'>
            <ul className='nav flex-column' >
                <div className='container-fluid'>
                        <li className='nav-item'>
                            <div className='row my-3'>
                                <button 
                                    className='btn btn-dark'
                                    value='home'
                                    onClick={ e => handleClick(e) }
                                >
                                    <i className="bi bi-house-door-fill d-block"></i> 
                                    Home
                                </button>
                            </div>
                        </li>
                        <li className='nav-item'>
                            <div className='row my-3'>
                                <button 
                                    className='btn btn-dark'
                                    value='dashboard'
                                    onClick={ e => handleClick(e) }
                                >
                                    <i className="bi bi-clipboard-data-fill d-block"></i> 
                                    Dashboard
                                </button>
                            </div>
                        </li>
                        <li className='nav-item '>
                            <div className='row my-3'>
                                <button 
                                    className='btn btn-dark'
                                    value='orders'
                                    onClick={ e => handleClick(e) }
                                >
                                    <i className="bi bi-file-earmark-text-fill d-block"></i>
                                    Orders
                                </button>
                            </div>
                        </li>
                        <li className='nav-item '>
                            <div className='row my-3'>
                                <button 
                                    className='btn btn-dark'
                                    value='products'
                                    onClick={ e => handleClick(e) }
                                >
                                    <i className="bi bi-cart-fill d-block"></i>
                                    Products
                                </button>
                            </div>
                        </li>
                        <li className='nav-item '>
                            <div className='row my-3'>
                                <button 
                                    className='btn btn-dark'
                                    value='customers'
                                    onClick={ e => handleClick(e) }
                                >
                                    <i className="bi bi-people-fill d-block"></i>
                                    Customers
                                </button>
                            </div>
                        </li>
                        <li className='nav-item '>
                            <div className='row my-3'>
                                <button 
                                    className='btn btn-dark'
                                    value='reports'
                                    onClick={ e => handleClick(e) }
                                >
                                    <i className="bi bi-bar-chart-fill d-block"></i>
                                    Reports
                                </button>
                            </div>
                        </li>
                        <li className='nav-item '>
                            <div className='row my-3'>
                                <button 
                                    className='btn btn-dark'
                                    value='integrations'
                                    onClick={ e => handleClick(e) }
                                >
                                    <i className="bi bi-layers-fill d-block"></i>
                                    Integrations
                                </button>
                            </div>
                        </li>
                </div>
            </ul>
        </div>
    </nav>
  )
}
