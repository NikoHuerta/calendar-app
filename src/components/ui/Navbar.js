import React from 'react'
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { startLogout } from '../../actions/auth';

export const Navbar = () => {

    const dispatch = useDispatch(); 
    const { name, rol } = useSelector(store => store.auth);
    const navigate = useNavigate();


    const handleLogout = () => {
        dispatch( startLogout() );
    }

    const handleAdmin = () => {
        console.log('Go admin zone!');
        navigate('/admin', { replace: true });
    }

    return (
        <div className='navbar navbar-dark bg-dark mb-4'>
            <span className='navbar-brand'>
                { name }
            </span>
            
            <form className="form-inline">
                <button 
                    className='btn btn-outline-info mr-sm-2'
                    onClick={ handleAdmin }
                    disabled={ (rol==='ADMIN_ROLE') ? false : true }
                >
                    <i className="fas fa-user-cog"></i>
                    <span> Admin Zone </span>
                </button>

                <button 
                    className='btn btn-outline-danger my-2 my-sm-0'
                    onClick={ handleLogout }
                >
                    <i className='fas fa-sign-out-alt'></i>
                    <span> Logout </span>
                </button>
            </form>
        </div>
    )
}
