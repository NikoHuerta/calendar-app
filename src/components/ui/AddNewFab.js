import React from 'react'
import { useDispatch } from 'react-redux';
import { eventRemoveActive } from '../../actions/calendarEvents';
import { uiOpenModal } from '../../actions/ui';
// FAB floating action button
export const AddNewFab = () => {

    const dispatch = useDispatch();

    const handleClickNew = () => {
        dispatch( eventRemoveActive() );
        dispatch( uiOpenModal() );
    };

    return (
        <button
            className='btn btn-primary fab'
            onClick={ handleClickNew }
        >
            <i className='fas fa-plus'></i>
        </button>
    )
}
