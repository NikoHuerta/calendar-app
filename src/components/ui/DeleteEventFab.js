import React from 'react'
import { useDispatch, useSelector } from 'react-redux';
import { eventStartDelete } from '../../actions/calendarEvents';

export const DeleteEventFab = () => {

    const dispatch = useDispatch();
    const { activeEvent } = useSelector( state => state.calendar );

    const handleDelete = () => {
        dispatch( eventStartDelete(activeEvent) );
    }

    return (
        <button
            className='btn btn-danger fab-danger'
            onClick={ handleDelete }
        >
            <i className='fas fa-trash'></i>
            <span> Borrar Evento </span>
        </button>
    )
}
