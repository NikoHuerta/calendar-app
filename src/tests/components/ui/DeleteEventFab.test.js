import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';

import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { eventStartDelete } from '../../../actions/calendarEvents';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

const initState = {
    calendar: {
        activeEvent: '12345678'
    }
};
let store = mockStore( initState );
store.dispatch = jest.fn(); //solo importa que se dispare, por eso el mock

jest.mock('../../../actions/calendarEvents', () => ({
    eventStartDelete: jest.fn()
}));


const wrapper = mount(
    <Provider store={ store }>
        <DeleteEventFab />
    </Provider>
);

describe('Pruebas en componente <DeleteEventFab />', () => {

    beforeEach(()=>{
        store = mockStore(initState);
        jest.clearAllMocks();
    });

    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de llamar a la accion correctamente', () => {
        wrapper.find('button').prop('onClick')();
        expect(eventStartDelete).toHaveBeenCalledWith('12345678');
    });
});