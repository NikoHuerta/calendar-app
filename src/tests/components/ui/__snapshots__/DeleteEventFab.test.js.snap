// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <DeleteEventFab /> Debe de mostrarse correctamente 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Borrar Evento 
  </span>
</button>
`;
